src/Compilers/CSharp/Portable/Symbols/Source/SourceConstructorSymbolBase.cs

```diff
@@ -246,5 +246,40 @@ internal sealed override int CalculateLocalSyntaxOffset(int position, SyntaxTree
         protected abstract CSharpSyntaxNode GetInitializer();
 
         protected abstract bool IsWithinExpressionOrBlockBody(int position, out int offset);
+
+#nullable enable
+        protected sealed override bool HasSetsRequiredMembersImpl
+            => GetEarlyDecodedWellKnownAttributeData()?.HasSetsRequiredMembersAttribute == true;
+
+        internal sealed override (CSharpAttributeData?, BoundAttribute?) EarlyDecodeWellKnownAttribute(ref EarlyDecodeWellKnownAttributeArguments<EarlyWellKnownAttributeBinder, NamedTypeSymbol, AttributeSyntax, AttributeLocation> arguments)
+        {
+            if (arguments.SymbolPart == AttributeLocation.None)
+            {
+                if (CSharpAttributeData.IsTargetEarlyAttribute(arguments.AttributeType, arguments.AttributeSyntax, AttributeDescription.SetsRequiredMembersAttribute))
+                {
+                    var earlyData = arguments.GetOrCreateData<MethodEarlyWellKnownAttributeData>();
+                    earlyData.HasSetsRequiredMembersAttribute = true;
+
+                    if (ContainingType.IsWellKnownSetsRequiredMembersAttribute())
+                    {
+                        // Avoid a binding cycle for this scenario.
+                        return (null, null);
+                    }
+
+                    var (attributeData, boundAttribute) = arguments.Binder.GetAttribute(arguments.AttributeSyntax, arguments.AttributeType, beforeAttributePartBound: null, afterAttributePartBound: null, out bool hasAnyDiagnostics);
+
+                    if (!hasAnyDiagnostics)
```

Comment:
This is our standard pattern. We let normal binding handle it, rather than early binding, because early binding has restrictions that normal binding does not.